src/app/home/pages/ClustersPage/components/ClustersTable.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useEffect } from 'react';
+import { Button, Pagination, Level, LevelItem } from '@patternfly/react-core';
+import { Table, TableHeader, TableBody, sortable, classNames } from '@patternfly/react-table';
+import { LinkIcon } from '@patternfly/react-icons';
+import spacing from '@patternfly/react-styles/css/utilities/Spacing/spacing';
+import tableStyles from '@patternfly/react-styles/css/components/Table/table';
+import { usePaginationState, useSortState } from '../../../../common/duck/hooks';
+import { getClusterInfo } from '../helpers';
+import StatusIcon from '../../../../common/components/StatusIcon';
+import ClusterActionsDropdown from './ClusterActionsDropdown';
+import IconWithText from '../../../../common/components/IconWithText';
+
+// TODO add prop types interface
+const ClustersTable = ({
+  clusterList,
+  associatedPlans,
+  migMeta,
+  removeCluster,
+  toggleAddEditModal,
+}) => {
+  const columns = [
+    { title: 'Name', transforms: [sortable] },
+    { title: 'Location', transforms: [sortable] },
+    { title: 'Associated plans', transforms: [sortable] },
+    { title: 'Status', transforms: [sortable] },
+    { title: '', columnTransforms: [classNames(tableStyles.tableAction)] },
+  ];
+
+  // TODO add type for cluster?
+  const getSortValues = (cluster) => {
+    const { clusterName, clusterUrl, associatedPlanCount, clusterStatus } = getClusterInfo(
+      cluster,
+      migMeta,
+      associatedPlans
+    );
+    return [clusterName, clusterUrl, associatedPlanCount, clusterStatus, ''];
+  };
+
+  const { sortBy, onSort, sortedItems } = useSortState(clusterList, getSortValues);
+  const { currentPageItems, setPageNumber, paginationProps } = usePaginationState(sortedItems, 10);
```

Comment:
Really cool use of hooks here. The composition pattern is very clean & it is great to be able to pull from common hooks across the entire app. Great Job!

Also, +1 for consistency between the pages & your pattern of extracting derived state/data formatting to helper functions.